src/core/service.c

```diff
@@ -1251,23 +1265,96 @@ static int service_coldplug(Unit *u) {
         return 0;
 }
 
+static int connect_unix_probe(Service *s, OpenFile *of, int ofd) {
+        union sockaddr_union addr;
```

Comment:
We need to set the family to `AF_UNIX`:

```
union sockaddr_union sa = {
        .un.sun_family = AF_UNIX,
};
```

Reply:
Doing this might also change the error returned by `connect()`, so check again exactly what error it returns if the socket type is incorrect.